### ISSUE TITLE
no-unused-declaration false positive for global augmentations

### ISSUE THREAD

**Issue by danvk** [cartant/eslint-plugin-etc#18]
You can tell TypeScript that you'd like to [globally augment `window` via `declare global`](https://github.com/microsoft/TypeScript/issues/33128):

```ts
declare global {
  interface Window {
    myGlobalVar?: string;
  }
}
```

This makes the following code type safe:

```ts
declare global {
  interface Window {
    myGlobalVar?: string;
  }
}

window.myGlobalVar  // type is string | undefined
```

The no-unused-declaration rule flags this as unused, however:

```ts
declare global {
  interface Window {
    //      ~~~~~~ "Window" is not used; unused declarations are forbidden.
    //             eslintetc/no-unused-declaration
    myGlobalVar?: string;
  }
}
```
